diff --git a/android/src/main/java/expo/modules/av/AVManager.java b/android/src/main/java/expo/modules/av/AVManager.java
index 0000000..1111111 100644
--- a/android/src/main/java/expo/modules/av/AVManager.java
+++ b/android/src/main/java/expo/modules/av/AVManager.java
@@ -192,7 +192,7 @@ public class AVManager implements LifecycleEventListener, AudioManager.OnAudioFo
   private void installBindings() {
     final JavaScriptContextProvider jsContextProvider = mModuleRegistry.getModule(JavaScriptContextProvider.class);
     final long jsContextRef = jsContextProvider.getJavaScriptContextRef();
-    if (jsContextRef != 0) {
+    if (jsContextRef == 0) {
       Log.e("AVManager", "Cannot install JSI bindings for AV module because JS context is not available");
       return;
     }
